From a3006d23a58d825dcaa4863ce81a454e151d038d Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Mon, 21 May 2018 16:45:02 +0200
Subject: [PATCH] HACK: only build and install localedef

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>

Upstream: https://git.pengutronix.de/cgit/ptxdist/plain/patches/localedef-glibc-2.27/0001-HACK-only-build-and-install-localedef.patch?id=47116f66f411d4dadfce42c2fdd6d41b351ccfd4
Signed-off-by: Peter Seiderer <ps.report@gmx.net>
[Romain: rebase on 2.38]
Signed-off-by: Romain Naour <romain.naour@gmail.com>
[yann.morin.1998@free.fr: rebase for 2.39]
Signed-off-by: Yann E. MORIN <yann.morin.1998@free.fr>
[Bernd: rebase for 2.44]
Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 Rules           | 14 ++++++++++----
 locale/Makefile |  3 ---
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/Rules b/Rules
index 71495028fb..5e1243e44d 100644
--- a/Rules
+++ b/Rules
@@ -263,10 +263,16 @@ binaries-shared-notests = $(filter-out $(binaries-pie) $(binaries-static), \
 				       $(binaries-all-notests))
 
 ifneq "$(strip $(binaries-shared-notests))" ""
-$(addprefix $(objpfx),$(binaries-shared-notests)): %: %.o \
-  $(sort $(filter $(common-objpfx)lib%,$(link-libc))) \
-  $(addprefix $(csu-objpfx),start.o) $(+preinit) $(+postinit)
-	$(+link)
+$(addprefix $(objpfx),$(binaries-shared-notests)): %: %.o
+	$(CC) -o $@ \
+	      $(sysdep-LDFLAGS) $(LDFLAGS) $(LDFLAGS-$(@F)) \
+	      $(combreloc-LDFLAGS) $(relro-LDFLAGS) $(hashstyle-LDFLAGS) \
+	      $(filter-out $(addprefix $(csu-objpfx),start.o \
+						     $(start-installed-name))\
+			   $(+preinit) \
+			   $(link-extra-libs) \
+			   $(common-objpfx)libc% $(+postinit),$^) \
+	      $(link-extra-libs)
 endif
 
 ifneq "$(strip $(binaries-shared-tests))" ""
diff --git a/locale/Makefile b/locale/Makefile
index 25f9b5bbce..5cbcd34bd3 100644
--- a/locale/Makefile
+++ b/locale/Makefile
@@ -77,7 +77,6 @@ aux = \
   xlocale \
   # aux
 others = \
-  locale \
   localedef \
   # others
 #others-static = \
@@ -85,7 +84,6 @@ others = \
 # locale \
 # # others-static
 install-bin = \
-  locale \
   localedef \
   # install-bin
 extra-objs = \
@@ -97,7 +95,6 @@ extra-objs = \
 generated += C-translit.h
 before-compile += $(objpfx)C-translit.h
 
-extra-libs = libBrokenLocale
 extra-libs-others = $(extra-libs)
 
 libBrokenLocale-routines = broken_cur_max
-- 
2.50.1

